grammars/ruby.cson

```diff
@@ -351,7 +351,8 @@
   }
   {
     'match': '''(?x)
-      \\b(
+      (\\b|(?<!\\d[ \\t]|\\d)-)
```

Comment:
Both of those have the same problems you just demonstrated with the double spaces.  It also gets more complicated with parentheses: `(3) - 4`.  So I think reverting 0a886fc is the correct choice here, as `-` is still technically an operator in these cases.